mlir/utils/widgets/rockE2E.sh

```diff
@@ -0,0 +1,433 @@
+                        ## E.g. -rand_type and -arch %arch/gfx1030
+                        param=${configCLToVar[$key]}
+                        configArr[$param]=$value
+                    fi
+                fi
+            fi
+        fi
+    done
+}
+
+## Match the given config entry in the config_db
+## and insert the entry into the db if not exist
+function match_and_insert() {
+    ## $1: config entry
+    entry=$1
+    entry=$(echo $entry | xargs)
+    foundOne=0
+    for db in ${DB_ARR[@]};
+    do
+        if [ ! -f $db ];then
+            db=${CONVDB_DIR}/$db.toml
+        fi
+        grep "config =" ${db} > db.tmp
+        dbN=1
+        while read -r line;
+        do
+            entry_db=$(canonicalize_config "$line")
+            if [[ "$entry" == "$entry_db" ]];then
+                printf "    \xE2\x9C\x94 Match config $dbN in ${db##*/}\n"
+                foundOne=1
+            fi
+            ((dbN++))
+        done < db.tmp
+        ## If not found in the database, insert if MODE=insert
+        if [[ $foundOne -eq 0 ]] && [[ "$MODE" == "insert" ]];then
+            echo "    -> Insert new config into ${db##*/}"
+            echo "" >> $db
+            echo "[[suite.test]]" >> $db
+            echo "config = \"$entry\"" >> $db
+        fi
+        rm db.tmp
+    done
+    ## Not found in any database
+    if [[ $foundOne -eq 0 ]] && [[ "$MODE" == "search" ]];then
+        printf "    \xE2\x9D\x8C Not found in any database\n"
+    fi
+}
+
+
+function check_database() {
+    ## Set database to convDB if not set at command line
+    DATABASE=${DATABASE_OPTION:-$convDB}
+    IFS=',' read -r -a DB_ARR <<< $DATABASE
+    err=0
+    for db in ${DB_ARR[@]};
+    do
+        if [ ! -f $db ] && [ ! -f ${CONVDB_DIR}/$db.toml ];then
+            err=1
+            echo -e ${COL_ERR}"ERROR: $db does not exist" | stderr
+        fi
+    done
+    [[ $err -eq 0 ]] || exit 0
+    if [[ "$MODE" == "insert" ]] && [[ "$DATABASE" =~ "," ]];then
+        echo -e ${COL_ERR}"ERROR: Only one database is allowed"`
+                        ` "in insert mode" | stderr
+        exit 1
+    fi
+}
+
+function check_inputs() {
+    if [[ "${INPUT_OPTION}" != "" ]]; then
+        process_input
+    elif [[ "${CONFIG_STR}" != "" ]];then
+        entry=$(canonicalize_config "${CONFIG_STR}")
+        match_and_insert "$entry"
+    else
+        echo -e ${COL_ERR}"ERROR: No input. Must provide input by" `
+                         `"--input= or --config=" | stderr
+        exit 1
+    fi
+}
+
+
+function check_formats() {
+    ## Overwrite the formats for insert mode as the default formats
+    if [[ "$MODE" == "insert" ]] || [[ ${#FORMATS[@]} -eq 0 ]];then
+        FORMATS=(no-dt no-dir no-layouts)
+    fi
+}
+
+
+PROG=${0##*/}
+PROG_DIR=$(cd ${0%/*} && pwd -P)
+ROCMLIR_DIR=$(cd ${PROG_DIR}; git rev-parse --show-toplevel || echo "${HOME}/rocMLIR/")
+CONVDB_DIR=${ROCMLIR_DIR}/mlir/test/e2e
+
+convDB="${CONVDB_DIR}/MixedConvLayouts.toml"
```

Comment:
Nit: Why not just `for i in "${CONVDB_DIR}/*.toml"` and then, for instance, dropping the leading comma. This makes the script less error-prone in the future.

Reply:
Ok, wait, hold on, is this meant to be only for convolutions?

Reply:
For now, yes. 
It should easy to extended for Gemm. Or a similar script for Gemm.

Reply:
I'll admit I don't entirely understand the point of this script

Reply:
When we want to expand the E2E tests, this script can help check if a given test is already in one of the toml configuration files.

Reply:
(But that can be a followup PR)